Cargo.toml

```diff
@@ -8,6 +8,7 @@ repository = "https://github.com/viamrobotics/micro-rdk"
 resolver = "2"
 version = "0.1.6"
 rust-version = "1.71"
+links = "micro_rdk"
```

Comment:
This argument is meant for linking native libraries at least according to the doc. Is that usage explain somewhere? feels weird to have to link against ourselves or am I missing something?

Reply:
Yes, I should have said more in my comment above.

That is the documented purpose of this field per the cargo docs. But it appears that `embuild` uses this field somewhat differently, as a key to understand how and when to propagate toolchain settings. See for instance, the rather cryptic comment here: https://github.com/esp-rs/embuild/blob/master/src/build.rs#L392. It took me a while to figure out that they were talking about https://doc.rust-lang.org/cargo/reference/manifest.html#the-links-field, which is a little thin on details w.r.t. the effects of specifying it. What gives me some hope that this is not, in fact, as weird as it looks (or is at least no weirder than what is already going on in the esp-idf ecosystem), is that this is what `esp-idf-svc` itself does, per https://github.com/esp-rs/esp-idf-svc/blob/master/Cargo.toml#L13, in order to propagate its settings / args. As far as I can see there is no actual `esp_idf_svc` native lib in play: the `esp_idf_svc` project is entirely rust. Perhaps it would have been better if `embuild` had used custom package metadata to do this, like we did for micro-rdk modules: https://github.com/viam-labs/micro-rdk-esp32-sensor-examples/blob/main/Cargo.toml#L17-L18.

In any event, if you think this is a step to far (and I won't blame you in the least), we can back down to requiring that projects retain an explicit dependency on `esp-idf-svc` in their `[dependencies]` section.

Reply:
Also, I'd like to add that all of the above is per my best current understanding of how these things work and what `embuild` is doing. It is entirely possible I've got it wrong or that there is another better way.

Reply:
I don't think we should do this it feels incorrect. We should be able to propagate the proper env with embuild.